uberfire-structure/uberfire-structure-backend/src/main/java/org/guvnor/structure/backend/repositories/RepositoryServiceImpl.java

```diff
@@ -450,6 +450,8 @@ protected void doRemoveRepository(final OrganizationalUnit orgUnit,
             //Remove reference to Repository from Organizational Units
             for (Repository repository : orgUnit.getRepositories()) {
                 if (repository.getAlias().equals(alias)) {
+                    configStorage.deleteRepository(alias);
```

Comment:
This method should not be called here (and I think it shouldn't be needed anywhere else). Here, the repository is only marked for deletion by the `organizationalUnitService` (with lock acquired), and later the `FileSystemDeleteWorker#removeRepository(final Repository repo)` will remove the whole repository directory at once, which should remove the change requests along with it.

Reply:
Thanks for pointing this out! I guess  `FileSystemDeleteWorker` wasn't there yet when I changed this method the first time. Let me check if we still need it.